cache.go

```diff
@@ -399,6 +399,29 @@ func (cache *Cache) Stats() Stats {
 	}
 }
 
+// Resize the cache to hold at most n entries. If n is smaller than the current
+// size, entries are evicted to fit the new size.
+func (cache *Cache) Resize(n int) {
+	if n <= 0 {
+		panic("Must supply a positive capacity to Resize")
```

Comment:
Wondering if that's a little to dangerous for something that happens during runtime, post-init. Maybe returning an `error` would be gentler and let the caller decide how harsh they want to be.

Reply:
That sounds like a great idea to me.